test/adapter/json_api/relationship_test.rb

```diff
@@ -4,132 +4,204 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi
       class RelationshipTest < ActiveSupport::TestCase
+        class Author < ActiveModelSerializers::Model
+          attr_accessor :id, :name, :blog, :posts
+        end
+        class AuthorSerializer < ActiveModel::Serializer
+          type :authors
+          attributes :id, :name
+
+          has_one :blog
+          has_one :blog_without_serializer
+          has_many :posts
+        end
+        class Blog < ActiveModelSerializers::Model
+          attr_accessor :id, :name
+        end
+        class BlogWithoutSerializer < Blog; end
+        class BlogSerializer < ActiveModel::Serializer
+          type :blogs
+          attributes :id, :name
+        end
+
         setup do
-          @blog = Blog.new(id: 1)
-          @author = Author.new(id: 1, name: 'Steve K.', blog: @blog)
-          @serializer = BlogSerializer.new(@blog)
-          ActionController::Base.cache_store.clear
+          @serializable_resource_options = {} # adapter.instance_options
+        end
+
+        def build_author_relationship(author_attributes, association_name)
```

Comment:
Just that normally these are created by the serializer using these as defaults in the reflection initializer, but here I am mutating them, which changes them for all the tests, and dup and deep_dup didn't work.  So it's a hack to make the serializer behave in each test as if it were being defined for the first time.  

B mobile phone

> On Aug 18, 2016, at 6:53 AM, L. Preston Sego III <notifications@github.com> wrote:
> 
> In test/adapter/json_api/relationship_test.rb:
> 
> >          setup do
> > -          @blog = Blog.new(id: 1)
> > -          @author = Author.new(id: 1, name: 'Steve K.', blog: @blog)
> > -          @serializer = BlogSerializer.new(@blog)
> > -          ActionController::Base.cache_store.clear
> > +          @serializable_resource_options = {} # adapter.instance_options
> > +        end
> > +
> > +        def build_author_relationship(author_attributes, association_name)
> I think it's pretty straight forward, usage wise.
> 
> though, why do you need to reset the reflection stuff?
> is that for isolation or something else?
> my concern is that if something is happening to those normally that is being avoided here, it could reveal a hole in testing.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>